src/lib/utility.js

```diff
@@ -124,6 +124,57 @@ module.exports = (requires) => {
   utilities.filter = (str) => {
     return str.replace('%prefix', config.prefix);
   };
+
+  //for finding the command object for either the keyword or an alias 
+  utilities.getCommand = function(keyword) 
+  { 
+    let commands = info.commands;
+    if(commands[keyword] && typeof commands[keyword].getAction() === 'function') { 
+      return commands[keyword]; 
+    } else { 
+      //didn't find command 
+      for(let index in commands) { 
+        if(commands[index] && typeof commands[index] === 'object'
+           && commands[index].getAlias().indexOf(keyword) > -1) { 
+          return commands[index]; 
+        } 
+      } 
+    } 
+    return null; 
+  };
+
+  utilities.getPermLevel = function(details) {
+    let db = info.db;
+    let roles = details.member ? details.member.roles : [];
+    let permLevel = 'public';
+    if(details.isAdministrator) {
+      permLevel = 'private';
+    } else {
+      permLevel = db.findPerm(details.userId, roles).then(perm => {
+        if(perm != null) {
+          return perm._id;
+        }
+        return 'public';
+      }).catch(console.log);
+    }
+    return permLevel;
+  };
+
+  utilities.hasPermission = function(commandLevel, userLevel) {
```

Comment:
Arrow, I'll check the logic for this when you finish up the other changes but it might not work with the `high` and `low` roles.